source/renderer/app/components/staking/stake-pools/StakePoolsList.js

```diff
@@ -1,164 +1,91 @@
-   * these 3 props need to be passed, as it's the parent who will control
-   * which list is active and prevent multiple Tooltips to be displayed
-   *
-   */
-  listName?: string,
-  isListActive?: boolean,
-  setListActive?: Function,
-};
-
-type State = {
-  highlightedPoolId?: ?number,
-  isPreloading: boolean,
 };
 
-const initialState = {
-  highlightedPoolId: null,
-  isPreloading: true,
-};
-
-@observer
-export class StakePoolsList extends Component<Props, State> {
-  static defaultProps = {
-    isListActive: true,
-    showWithSelectButton: false,
-  };
-
-  constructor(props: Props) {
-    super(props);
-    window.addEventListener('resize', this.handleResize);
-  }
-
-  state = { ...initialState };
-
-  // We need to track the mounted state in order to avoid calling
-  // setState promise handling code after the component was already unmounted:
-  // Read more: https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html
-  _isMounted = false;
-
-  componentDidMount() {
-    this._isMounted = true;
-    setTimeout(() => {
-      if (this._isMounted) this.setState({ isPreloading: false });
-    }, 0);
-  }
-
-  componentWillUnmount() {
-    this._isMounted = false;
-    window.removeEventListener('resize', this.handleClose);
-  }
-
-  handleResize = () =>
-    debounce(this.handleClose, 200, { leading: true, trailing: false });
-
-  searchInput: ?HTMLElement = null;
-
-  getIsHighlighted = (id: string) =>
-    this.props.isListActive !== false && id === this.state.highlightedPoolId;
-
-  handleOpenThumbnail = (highlightedPoolId: number) => {
-    const { isListActive, setListActive, listName } = this.props;
-    if (isListActive === false && setListActive) setListActive(listName);
-    return this.setState({
-      highlightedPoolId,
-    });
-  };
-
-  handleClose = () => {
-    this.setState({
-      ...initialState,
-      isPreloading: false,
-    });
-  };
-
-  handleSelect = (stakePoolId: number) => {
-    const { onSelect } = this.props;
-    const selectedPoolId =
-      this.props.selectedPoolId === stakePoolId ? null : stakePoolId;
-    if (onSelect) {
-      onSelect(selectedPoolId);
-    }
-  };
-
-  render() {
-    const {
-      currentTheme,
-      highlightOnHover,
-      onOpenExternalLink,
-      showSelected,
-      showWithSelectButton,
-      stakePoolsList,
-      selectedPoolId,
-      containerClassName,
-      numberOfRankedStakePools,
-      disabledStakePoolId,
-      listName,
-    } = this.props;
-    const { isPreloading } = this.state;
-    const componentClasses = classNames([styles.component, listName]);
-
-    if (stakePoolsList.length > PRELOADER_THRESHOLD && isPreloading)
-      return (
-        <div className={styles.preloadingBlockWrapper}>
-          <LoadingSpinner big />
-        </div>
-      );
+/**
+ * Utility function to programmatically hide all pop overs
+ * This is used to hide the pool tooltips on scrolling the list
+ */
+function hideAllPopOvers() {
```

Comment:
@domenkozar  Can you change this to an arrow function or we should use this as a new approach?

Reply:
You wanted to tag @DominikGuzei here I believe :-)

Reply:
hehe that is true 😄

Reply:
@tomislavhoracek in this case there is no reason to use an arrow function as this is a global function and no context binding is required 😉 I don't think we have a policy of "only use arrow functions" and I would rather recommend the opposite: use arrow functions if it makes sense (e.g: inline definitions, context binding etc.)

Reply:
Agreed @DominikGuzei 👍